src/Bigcommerce/Api/Connection.php

```diff
@@ -8,455 +8,469 @@
+     * if the last request was not an error.
+     */
+    public function getLastError()
+    {
+        return $this->lastError;
+    }
+
+    /**
+     * Recursively follow redirect until an OK response is recieved or
+     * the maximum redirects limit is reached.
+     *
+     * Only 301 and 302 redirects are handled. Redirects from POST and PUT requests will
+     * be converted into GET requests, as per the HTTP spec.
+     */
+    private function followRedirectPath()
+    {
+        $this->redirectsFollowed++;
+
+        if ($this->getStatus() == 301 || $this->getStatus() == 302) {
+
+            if ($this->redirectsFollowed < $this->maxRedirects) {
+
+                $location = $this->getHeader('Location');
+                $forwardTo = parse_url($location);
+
+                if (isset($forwardTo['scheme']) && isset($forwardTo['host'])) {
+                    $url = $location;
+                } else {
+                    $forwardFrom = parse_url(curl_getinfo($this->curl, CURLINFO_EFFECTIVE_URL));
+                    $url = $forwardFrom['scheme'] . '://' . $forwardFrom['host'] . $location;
+                }
+
+                $this->get($url);
+
+            } else {
+                $errorString = "Too many redirects when trying to follow location.";
+                throw new NetworkError($errorString, CURLE_TOO_MANY_REDIRECTS);
+            }
+        } else {
+            $this->redirectsFollowed = 0;
+        }
+    }
+
+    /**
+     * Make an HTTP GET request to the specified endpoint.
+     */
+    public function get($url, $query = false)
+    {
+        $this->initializeRequest();
+
+        if (is_array($query)) {
+            $url .= '?' . http_build_query($query);
+        }
+
+        curl_setopt($this->curl, CURLOPT_CUSTOMREQUEST, 'GET');
+        curl_setopt($this->curl, CURLOPT_URL, $url);
+        curl_setopt($this->curl, CURLOPT_POST, false);
+        curl_setopt($this->curl, CURLOPT_PUT, false);
+        curl_setopt($this->curl, CURLOPT_HTTPGET, true);
+        curl_exec($this->curl);
+
+        return $this->handleResponse();
+    }
+
+    /**
+     * Make an HTTP POST request to the specified endpoint.
+     */
+    public function post($url, $body)
+    {
+        $this->addHeader('Content-Type', $this->getContentType());
+
+        if (!is_string($body)) {
+            $body = json_encode($body);
+        }
+
+        $this->initializeRequest();
+
+        curl_setopt($this->curl, CURLOPT_CUSTOMREQUEST, 'POST');
+        curl_setopt($this->curl, CURLOPT_URL, $url);
+        curl_setopt($this->curl, CURLOPT_POST, true);
+        curl_setopt($this->curl, CURLOPT_PUT, false);
+        curl_setopt($this->curl, CURLOPT_HTTPGET, false);
+        curl_setopt($this->curl, CURLOPT_POSTFIELDS, $body);
+        curl_exec($this->curl);
+
+        return $this->handleResponse();
+    }
+
+    /**
+     * Make an HTTP HEAD request to the specified endpoint.
```

Comment:
There are several other instances of this. It's not a hard requirement for merge, but it would be great to generate documentation for this using phpdoc.